### ISSUE TITLE
analyze.spectral_reflectance breaks if there are empty ROIs

### ISSUE THREAD

**Issue by rayn-alex** [danforthcenter/plantcv#1450]
**Describe the bug**
analyze.spectral_reflectance breaks if there are empty ROIs. I already looked into the code and I think the problem is that the labelled mask levels and number of objects does not match. This will lead to empty sub masks which don't allow the calculation of min/max in the downstream analysis.

I think this is the culprit statement, which breaks the next steps if the mask is only zeros because wavelength_data will be empty:
`wavelength_data = array_data[np.where(mask > 0)]`
(in file spectral_reflectance.py L57)

I tested this with artificially by adding useless ROIs to a previously analysable image and analyze.spectral_reflectance did not work anymore.

**To Reproduce**
1. Run analyse.spectral_reflectance() using a labelled mask/n_label which was created with empty ROIs.
2. See error

**Expected behavior**
Run function without error.

**Local environment (please complete the following information):**
 - OS: Arch Linux
 - Environment: conda virtual env
 - PlantCV Version 4.0

**Additional context**
Error message

```
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
Cell In[21], line 2
      1 pcv.params.sample_label = "plant"
----> 2 spectral_hist = pcv.analyze.spectral_reflectance(hsi=spectral_array, labeled_mask=labeled_objects, n_labels=n_obj)

File ~/anaconda3/envs/plantcv4-dev/lib/python3.10/site-packages/plantcv/plantcv/analyze/spectral_reflectance.py:31, in spectral_reflectance(hsi, labeled_mask, n_labels, label)
     28 if label is None:
     29     label = params.sample_label
---> 31 _ = _iterate_analysis(img=hsi, labeled_mask=labeled_mask, n_labels=n_labels, label=label, function=_analyze_spectral)
     32 spectral_chart = outputs.plot_dists(variable="wavelength_means")
     33 _debug(visual=spectral_chart, filename=os.path.join(params.debug_outdir, str(params.device) + '_mean_reflectance.png'))

File ~/anaconda3/envs/plantcv4-dev/lib/python3.10/site-packages/plantcv/plantcv/_helpers.py:178, in _iterate_analysis(img, labeled_mask, n_labels, label, function, **kwargs)
    176 for i in range(1, n_labels + 1):
    177     submask = np.where(mask_copy == i, 255, 0).astype(np.uint8)
--> 178     img = function(img=img, mask=submask, label=f"{labels[i - 1]}_{i}", **kwargs)
    179 return img

File ~/anaconda3/envs/plantcv4-dev/lib/python3.10/site-packages/plantcv/plantcv/analyze/spectral_reflectance.py:61, in _analyze_spectral(img, mask, label)
     59 # Calculate mean reflectance across wavelengths
     60 wavelength_means = wavelength_data.mean(axis=0)
---> 61 max_per_band = wavelength_data.max(axis=0)
     62 min_per_band = wavelength_data.min(axis=0)
     63 std_per_band = wavelength_data.std(axis=0)

File ~/anaconda3/envs/plantcv4-dev/lib/python3.10/site-packages/numpy/core/_methods.py:41, in _amax(a, axis, out, keepdims, initial, where)
     39 def _amax(a, axis=None, out=None, keepdims=False,
     40           initial=_NoValue, where=True):
---> 41     return umr_maximum(a, axis, None, out, keepdims, initial, where)

ValueError: zero-size array to reduction operation maximum which has no identity
```

**Comment by rayn-alex** [comment]
I can fix that but wanted to be sure that this something new and it needs to be fixed. 

One can argue why would you want to analyse empty ROIs but if you look at time series and some plants in a grid appear later than others, it is important that the analysis does not break in the beginning ;)

**Comment by HaleySchuhl** [comment]
Thanks for opening this issue @rayn-alex , 100% agree we should handle empty replicates gracefully in `analyze.spectral_reflectance` just as we are doing with `analyze.size`.

**Comment by JSCG95** [comment]
Hi everyone!
I'm having the same issue reported when `analyze.spectral_reflectance` I'm not able to run an analysis per object when I have an empty ROI within my grid. That is not a problem with` analyze.size` as @HaleySchuhl mentioned.
 
Any alternative around this bug to run a spectral analysis for each object in my grid?

**Comment by nfahlgren** [comment]
Thanks @rayn-alex! This has been merged and is available by installing PlantCV from the source code, and will be included in the v4.3 release
